lib/logstash/inputs/kafka.rb

```diff
@@ -354,4 +357,16 @@ def set_sasl_config(props)
 
     props.put("sasl.kerberos.service.name",sasl_kerberos_service_name) unless sasl_kerberos_service_name.nil?
   end
+
+  def timestamp_ms
+    (Time.now.to_f * 1000).to_i
+  end
+
+  def has_to_commit?(last_commit_time)
+    # If auto_commit is enable we just leave the commit to the client library on poll and close actions
+    return false if @enable_auto_commit == "false"
+
+    # If auto_commit is disable, we need to commit, we will do it depending on the manual_commit_interval option
+    @manual_commit_interval_ms <= 0 || (last_commit_time + @manual_commit_interval_ms) < timestamp_ms
```

Comment:
for clarity, we can change the big conditional into two operations:

```ruby
  def has_to_commit?(last_commit_time)
    # auto_commit is enabled so we just leave the commit to the client library on poll and close actions
    return false if @enable_auto_commit == "true"

    # auto_commit is disabled but interval committing is disabled as well, so commit on every poll
    return true if @manual_commit_interval_ms <= 0

    # auto_commit is disabled and an interval is set, so let's check if enough time passed since last commit
    (last_commit_time + @manual_commit_interval_ms) < current_timestamp_ms
  end
```